main.rb

```diff
@@ -1,29 +1,208 @@
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    print "\nEnter your choice: "
+  end
+  
+  def insert_student(app, age)
+    print 'Name [Input the name]: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp
+    case parent_permission
+    when 'Y'
+      app.create_student(name, age, nil, parent_permission: true)
+    when 'N'
+      app.create_student(name, age, nil, parent_permission: false)
+    else
+      puts 'Invalid choice. Please try again'
+    end
+    puts "\nPerson created successfully\n"
+  end
+  
 
-rental1 = Rental.new('2023-07-27', book1, student1)
-rental2 = Rental.new('2022-06-18', book2, student2)
-rental3 = Rental.new('2021-05-09', book2, student1)
-rental4 = Rental.new('2020-04-13', book1, student2)
+  def insert_teacher(app, age)
+    if age_zero?(age)
+        return
+      end    
+    print 'Give me the name [Input the name]: '
+    name = gets.chomp
+    print 'Give me the specialization [Input the specialization]: '
+    spec = gets.chomp
+    app.create_teacher(name, age, spec)
+    puts "\n"
+    puts 'Person created successfully'
+    puts "\n"
+  end
 
-puts 'The classroom1 has these students:'
-classroom1.students.each { |student| puts student.name }
+  def age_zero?(age)
+    if age.zero?
+      puts "invalid age. Please try again\n"
+      true
+    else
+      false
+    end
+  end
+  
+  def insert_person(app)
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    choice2 = gets.chomp.to_i
+    case choice2
+    when 1
+      print 'You want to create a student. Give me the age [Input the number for age]: '
+      age = gets.chomp.to_i
+      age_zero?(age) ? return : insert_student(app, age)
+    when 2
+      print 'You want to create a teacher. Give me the age [Input the number for age]: '
+      age = gets.chomp.to_i
+      age_zero?(age) ? return : insert_teacher(app, age)
+    else
+      puts 'Invalid choice. Please try again'
+    end
+  end
+  
+  def insert_book(app)
+    print 'You want to create a book. Give me the title [Input the title]: '
+    title = gets.chomp
+    print 'Give me the author [Input the author]: '
+    author = gets.chomp
+    app.create_book(title, author)
+    puts "\n"
+    puts 'Book created successfully'
+    puts "\n"
+  end
+  
+  def insert_rental(app)
+    if app.people.empty?
+      puts "No people in the app yet. Add some people first to make a rental\n"
+      return
+    elsif app.books.empty?
+      puts "No books in the app yet. Add some books first to make a rental\n"
+      return
+    end
+    puts 'Select a book from the following list by number'
+    show_books_with_index(app)
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not id)'
+    show_people_with_index(app)
+    people_index = gets.chomp.to_i
+    print 'Give me the date [Input the date in format YYYY/MM/DD]: '
```

Comment:
- [ ] As per the project requirements, your `app.rb` file should serve as the entry point to your console app and should contain methods to perform the following tasks:
    
    - Display a list of all books
    - Display a list of all people
    - Create a new person (either a teacher or a student, not a plain `Person`)
    - Create a new book
    - Create a new rental
    - Display a list of all rentals for a specific person ID.
  
  However, I have observed that most of these methods are currently present in your `main.rb` file instead of the `app.rb`. Please make sure your app still works as expect while making the changes :pray: